File: pets/update-images.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'images-data.js');
let content = fs.readFileSync(filePath, 'utf8');

// Extract the JSON part
const match = content.match(/const petImagesData = ({[\s\S]*?})(;|\s*$)/);
if (!match) {
  console.error('Could not parse petImagesData');
  process.exit(1);
}

const jsonStr = match[1];
const data = eval('(' + jsonStr + ')');

const newEntry = {
  date: '2026-03-09',
  time: '01:00',
  img: '/images/pets/20260309-010000-BD0673744.jpg',
  type: 'other',
  boxedImg: '/images/pets/annotated/20260309-010000-BD0673744.jpg',
  analysis: 'image视觉识别和YOLO目标检测均未检测到人/猫/狗。图片模糊，整体呈灰白色调，无法识别具体内容。',
  petIdentity: null,
  yoloClasses: [],
  decisionSource: 'both'
};

// Add to 2026-03-09 array
if (!data.images['2026-03-09']) {
  data.images['2026-03-09'] = [];
}
data.images['2026-03-09'].unshift(newEntry);

// Recalculate stats
let personCount = 0, catCount = 0, dogCount = 0;
Object.values(data.images).forEach(arr => {
  arr.forEach(img => {
    if (img.type === 'person') personCount++;
    else if (img.type === 'cat') catCount++;
    else if (img.type === 'dog') dogCount++;
  });
});

data.stats = {
  totalImages: Object.values(data.images).reduce((sum, arr) => sum + arr.length, 0),
  personCount,
  catCount,
  dogCount
};

const newContent = 'const petImagesData = ' + JSON.stringify(data, null, 2) + ';';
fs.writeFileSync(filePath, newContent);

console.log('Updated successfully!');
console.log('Stats:', JSON.stringify(data.stats));
console.log('Entry count for 2026-03-09:', data.images['2026-03-09'].length);

File: pets/update_images.py
#!/usr/bin/env python3
import json
import re
import sys

with open("images-data.js", "r") as f:
    content = f.read()

# Find the JSON part
match = re.search(r'const petImagesData = (\{.*\});$', content, re.DOTALL)
if not match:
    print("ERROR: Could not find JSON data")
    sys.exit(1)

json_str = match.group(1)

try:
    data = json.loads(json_str)
except json.JSONDecodeError as e:
    print(f"ERROR: JSON parse error: {e}")
    sys.exit(1)

new_entry = {
    "date": "2026-03-18",
    "time": "09:06",
    "img": "/images/pets/20260318-090616-BD0673744.jpg",
    "type": "other",
    "found": False,
    "boxedImg": None,
    "analysis": "image: 图像模糊，无法确定是否有人/猫/狗; YOLO: 无检测",
    "petIdentity": None,
    "yoloClasses": [],
    "decisionSource": "both"
}

# Insert at beginning of 2026-03-18 array
if "2026-03-18" not in data["images"]:
    data["images"]["2026-03-18"] = []
data["images"]["2026-03-18"].insert(0, new_entry)

# Recalculate stats
total = 0
person = 0
cat = 0
dog = 0
for date_key in data["images"]:
    for img in data["images"][date_key]:
        total += 1
        if img.get("type") == "person":
            person += 1
        elif img.get("type") == "cat":
            cat += 1
        elif img.get("type") == "dog":
            dog += 1

data["stats"] = {"totalImages": total, "personCount": person, "catCount": cat, "dogCount": dog}

# Write back
new_content = "const petImagesData = " + json.dumps(data, indent=2, ensure_ascii=False) + ";\nexport default petImagesData;"

with open("images-data.js", "w") as f:
    f.write(new_content)

print("Updated successfully")
print("Stats:", data["stats"])
